packages/lab/stories/date-picker/date-picker.stories.tsx

```diff
@@ -1,209 +1,1099 @@
+  );
+};
+
+const formatDateStringEsES = (
+  date: DateValue | null | undefined,
+  locale?: string,
+  options?: Intl.DateTimeFormatOptions,
+) => {
+  const preferredLocale = locale || getCurrentLocale();
+  const preferredTimeZone = options?.timeZone || getLocalTimeZone();
+  return date
+    ? new DateFormatter(preferredLocale, {
+        day: "2-digit",
+        month: "2-digit",
+        year: "numeric",
+        ...options,
+        timeZone: preferredTimeZone,
+      }).format(date.toDate(preferredTimeZone))
+    : "";
+};
+
+export const RangeWithLocale: StoryFn<DatePickerRangeProps> = (args) => {
+  const [selectedDate, setSelectedDate] = useState<DateRangeSelection | null>(
+    null,
+  );
+  const helperText = "Date format DD/MM/YYYY";
+  const [validationStatus, setValidationStatus] = useState<"error" | undefined>(
     undefined,
   );
+
+  return (
+    <FormField validationStatus={validationStatus}>
+      <FormLabel>Select a date</FormLabel>
+      <DatePicker
+        {...args}
+        selectionVariant={"range"}
+        selectedDate={selectedDate}
+        locale={"es-ES"}
+        timeZone={"Europe/Madrid"}
+        onSelectedDateChange={(newSelectedDate: DateRangeSelection | null) => {
+          console.log(
+            `Selected date range: ${formatDateRange(newSelectedDate)}`,
+          );
+          setSelectedDate(newSelectedDate);
+          const validationStatus = isValidDateRange(newSelectedDate)
+            ? undefined
+            : "error";
+          setValidationStatus(validationStatus);
+          args?.onSelectedDateChange?.(newSelectedDate);
+        }}
+      >
+        <DatePickerRangeInput
+          onDateValueChange={(newDateValue?: DateInputRangeValue) => {
+            const validationStatus =
+              isValidNumericDate(newDateValue?.startDate, "DD/MM/YYYY") &&
+              isValidNumericDate(newDateValue?.endDate, "DD/MM/YYYY")
+                ? undefined
+                : "error";
+            setValidationStatus(validationStatus);
+          }}
+          formatDate={formatDateStringEsES}
+          parse={parseDateStringEsES}
+          placeholder={"DD/MM/YYYY"}
+        />
+        <DatePickerOverlay>
+          <DatePickerRangePanel />
+        </DatePickerOverlay>
+      </DatePicker>
+      <FormHelperText>{helperText}</FormHelperText>
+    </FormField>
+  );
+};
+
+export const SingleBordered: StoryFn<DatePickerSingleProps> = (args) => {
   return (
     <DatePicker
       {...args}
-      selectedDate={selectedDate}
-      onSelectionChange={(_, date) => setSelectedDate(date as DateValue)}
-    />
+      selectionVariant="single"
+      onSelectedDateChange={(newSelectedDate) => {
+        console.log(`Selected date: ${formatDate(newSelectedDate)}`);
+        args?.onSelectedDateChange?.(newSelectedDate);
+      }}
+    >
+      <DatePickerSingleInput bordered />
+      <DatePickerOverlay>
+        <DatePickerSinglePanel
+          NavigationProps={{
```

Comment:
Not 100% sure whether the props name here would make it confusing, v.s. `CalendarNavigationProps`

Reply:
agreed, renamed